classes/image_populate_mfgtool.bbclass

```diff
@@ -0,0 +1,142 @@
+# Allow generation of mfgtool bundle
+#
+# The class provides the infrastructure for MFGTOOL generation and is tied to images. To generate
+# the bundle, the task populate_mfgtool must be called. For example:
+#
+# ,----[ Running populate_mfgtool for core-image-minimal image ]
+# | $: bitbake core-image-minimal -c populate_mfgtool
+# `----
+#
+# The class behavior is controlled through the MFGTOOL_SCRIPT variable. The MFGTOOL_SCRIPT variable
+# itself specifies a space-separated list of the script to enable. Following the script, you can
+# determine the behavior of each script by providing up to two order-dependent arguments, which are
+# separated by commas. You can omit any argument you like but must retain the separating commas. The
+# order is important and specifies the following:
+#
+#   1. Extra dependencies that should be added to the do_populate_mfgtool task, if the script is
+#      enabled.
+#   2. Extra binaries that should be added to the bundle, if the script is enabled.
+#
+# For example:
+#
+# ,----[ Defining bootloader.uuu.in script ]
+# | MFGTOOL_SCRIPT = "bootloader.uuu.in"
+# | MFGTOOL_SCRIPT[bootloader.uuu.in] = "dep1:do_deploy dep2:do_deploy,file1 file2"
+# `----
+#
+# The virtual/bootloader:do_deploy is added to do_populate_mfgtool dependencies and ${UBOOT_BINARY}
+# copied to the bundle, only if the script is enabled.
+#
+# During the mfgtool bundle generation, the uuu.in files are processed and some variables
+# replaced. The variables are:
+#
+#   - MACHINE
+#   - UBOOT_BINARY
+#   - SPL_BINARY
+#   - IMAGE_BASENAME
+#
+# Copyright 2022-2023 (C) O.S. Systems Software LTDA.
+#
+# SPDX-License-Identifier: MIT
+
+MFGTOOL_FILESPATH ??= " \
+    ${@base_set_filespath(["%s/mfgtool" % p for p in "${BBPATH}".split(":")] \
+                             + ["${FILE_DIRNAME}/${BP}/mfgtool", \
+                                "${FILE_DIRNAME}/${BPN}/mfgtool", \
+                                "${FILE_DIRNAME}/files/mfgtool"] \
+                          , d)} \
+"
+
+MFGTOOLDIR = "${WORKDIR}/mfgtool-${PN}"
+do_populate_mfgtool[dirs] = "${MFGTOOLDIR}"
+do_populate_mfgtool[cleandirs] = "${MFGTOOLDIR}"
+
+addtask populate_mfgtool after do_image_complete do_unpack before do_deploy
+do_populate_mfgtool[dirs] ?= "${DEPLOY_DIR_IMAGE} ${WORKDIR}"
+do_populate_mfgtool[nostamp] = "1"
+do_populate_mfgtool[recrdeptask] += "do_deploy"
+do_populate_mfgtool[depends] += "uuu-bin-native:do_populate_sysroot"
+
+python () {
+    # Process MFGTOOL_SCRIPT and handle its fields.
+    script_flags = d.getVarFlags('MFGTOOL_SCRIPT') or {}
+    script_depends = []
+    script_deploy_files = []
```

Comment:
I see that this is where naming is tricky. The problem is `script_depends` and `script_deploy_files` here are sets of sets, so it's hard to come up with an accurate name that is not cumbersome. Although it could be argued that `script_depends` would be clearly a single set for a single `script`, while `scripts_depends` would be the full 'set of sets'.

How about using `flags`, `depends`, and `deploy_files` here, and `script_depends` and `script_deploy_files` for a single `script`?